templates/fine-tune-llm_v2/cookbooks/customize_initial_weights_and_prompt_format/README.ipynb

```diff
@@ -0,0 +1,203 @@
+   "source": [
+    "# Customizing the prompt format\n",
+    "\n",
+    "## How prompt formatting works in `llmforge`\n",
+    "\n",
+    "Here's a quick rundown of how prompt formatting/ chat templating works right now: the training/validation data needs to be formatted in the OpenAI messages format. Each example has a \"messages\" entry consisting a conversation with \"system\", \"user\" and \"assistant\" roles. For each role, depending on the model, we add certain tokens as headers/footers along with a BOS token at the start of the conversation and an EOS token at the end of each assistant response. The prompt format can be specified in our YAML as a part of the `generation_config` for the model (the same format is used in our inference code):\n",
+    "\n",
+    "```yaml\n",
+    "generation_config:\n",
+    "  prompt_format:\n",
+    "    system: \n",
+    "    user: \n",
+    "    assistant:\n",
+    "    trailing_assistant:  # inference-only\n",
+    "    bos: # optional\n",
+    "    system_in_user: # optional\n",
+    "    default_system_message: # optional\n",
+    "```\n",
+    "\n",
+    "For the models in the [list of supported models](../../README.md#faqs), we have default generation config parameters. This means that `generation_config` need not be specified when you just want to finetune a model like  `meta-llama/Meta-Llama-3-8B-Instruct` directly. \n",
+    "\n",
+    "### Examples\n",
+    "For `meta-llama/Meta-Llama-3-8B`, we use the following prompt format:\n",
+    "```yaml\n",
+    "generation_config:\n",
+    "  prompt_format:\n",
+    "    system: \"<|start_header_id|>system<|end_header_id|>\\n\\n{instruction}<|eot_id|>\"\n",
+    "    user: \"<|start_header_id|>user<|end_header_id|>\\n\\n{instruction}<|eot_id|>\"\n",
+    "    assistant: \"\"\"<|start_header_id|>assistant<|end_header_id|>\\n\\n{instruction}<|eot_id|>\"\n",
+    "    trailing_assistant: \"<|start_header_id|>assistant<|end_header_id|>\\n\\n\" # inference-only \n",
+    "    bos: \"<|begin_of_text|>\"\n",
+    "    system_in_user: False\n",
+    "    default_system_message: \"\"\n",
+    "```\n",
+    "\n",
+    "For `mistralai/Mistral-7B`, we the below prompt format:\n",
+    "```yaml\n",
+    "generation_config:\n",
+    "  prompt_format:\n",
+    "    system: \"{instruction} + \"\n",
+    "    user: \"[INST] {system}{instruction} [/INST]\"\n",
+    "    assistant: \" {instruction}</s>\"\n",
+    "    trailing_assistant: \"\" # inference-only \n",
+    "    bos: \"<s>\"\n",
+    "    system_in_user: True\n",
+    "    default_system_message: \"\"\n",
+    "```\n",
+    "\n",
+    "You can find more information on each entry in the [config API reference](https://docs.anyscale.com/reference/finetuning-config-api). Each `system`, `user` and `asistant` entry must contain the format specifier `{instruction}` which will actually format the corresponding entry in `messages`. For example, if a training example was as follows while training Llama-3-8B:\n",
```

Comment:
I think you should move this paragraph up to concretize exactly what you mean by prompt formatting. For a lot of people this doc will be their first learning doc regarding prompt format. :D

Reply:
Done!